extensions/notebook/src/test/book/book.test.ts

```diff
@@ -103,58 +103,94 @@ describe('BookTreeViewProviderTests @UNSTABLE@', function() {
 			await fs.writeFile(notebook2File, '');
 			await fs.writeFile(notebook3File, '');
 			await fs.writeFile(markdownFile, '');
-			console.log('Temporary folders and files created.');
-			mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
+		});
+
+		it('should initialize correctly with empty workspace array', async () => {
+			const bookTreeViewProvider = new BookTreeViewProvider([], mockExtensionContext, false, 'bookTreeView');
+			await bookTreeViewProvider.initialized;
+		});
+
+		it('should initialize correctly with workspace containing non-book path', async () => {
 			let folder: vscode.WorkspaceFolder = {
-				uri: vscode.Uri.file(rootFolderPath),
+				uri: vscode.Uri.file(nonBookFolderPath),
 				name: '',
 				index: 0
 			};
-			console.log('Creating BookTreeViewProvider...');
-			bookTreeViewProvider = new BookTreeViewProvider([folder], mockExtensionContext.object, false, 'bookTreeView');
-			let tocRead = new Promise((resolve, reject) => bookTreeViewProvider.onReadAllTOCFiles(() => resolve()));
-			let errorCase = new Promise((resolve, reject) => setTimeout(() => resolve(), 5000));
-			await Promise.race([tocRead, errorCase.then(() => { throw new Error('Table of Contents were not ready in time'); })]);
-			console.log('BookTreeViewProvider successfully created.');
+			const bookTreeViewProvider = new BookTreeViewProvider([folder], mockExtensionContext, false, 'bookTreeView');
+			await bookTreeViewProvider.initialized;
 		});
 
-		it('should return all book nodes when element is undefined', async function (): Promise<void> {
-			const children = await bookTreeViewProvider.getChildren();
-			should(children).be.Array();
-			should(children.length).equal(1);
-			book = children[0];
-			should(book.title).equal(expectedBook.title);
+		it('should initialize correctly with workspace containing both book and non-book paths', async () => {
+			const book: vscode.WorkspaceFolder = {
+				uri: vscode.Uri.file(bookFolderPath),
+				name: '',
+				index: 0
+			};
+			const nonBook: vscode.WorkspaceFolder = {
+				uri: vscode.Uri.file(nonBookFolderPath),
+				name: '',
+				index: 0
+			};
+			const bookTreeViewProvider = new BookTreeViewProvider([book, nonBook], mockExtensionContext, false, 'bookTreeView');
+			await bookTreeViewProvider.initialized;
+			should(bookTreeViewProvider.books.length).equal(1, 'Expected book was not initialized');
```

Comment:
>should [](start = 3, length = 6)

I thought anthony got rid of should and we should be using assert instead of it.

Reply:
That was for ADS core only because we were already using the assert library. Extensions can use whatever they want as long as they're consistent. In this case notebook is using both so we should consolidate at some point but that's beyond the scope of this PR.